cms/djangoapps/contentstore/views/preview.py

```diff
@@ -213,10 +214,27 @@ def _preview_module_system(request, descriptor, field_data):
             "i18n": ModuleI18nService,
             "settings": SettingsService(),
             "user": DjangoXBlockUserService(request.user),
+            "partitions": PreviewPartitionService(user=request.user, course_id=course_id)
         },
     )
 
 
+class PreviewPartitionService(PartitionService):
+    """
+    A runtime mixin to allow the display and editing of visibility based on user partitions.
+    """
+    def get_course(self):
+        return modulestore().get_course(self._course_id)
+
+    def get_user_group_id_for_partition(self, user_partition_id):
```

Comment:
We have to override this method because now when running in Studio a partitions runtime service actually exists. Previously, this code would just return None when trying to get the service (and the code was robust to that):

https://github.com/edx/edx-platform/blob/eea25184521c4fe224adc676006f6d86edb8e215/common/lib/xmodule/xmodule/split_test_module.py#L195